### DISCUSSION TITLE
Provider configuration not present with configuration_aliases

### DISCUSSION THREAD

**Discussion by rquadling** [terraform-linters/tflint#1922] [Q&A] [answered]
We build modules that get reused throughout multiple projects. Maybe not ideal, but we can now add and expand on functionality in a repo, release it with a semantic version tag and move on. Consuming projects can then upgrade their projects to use the new version as and when they see fit.

All good so far.

Our issue is with tflint running in the module repo.

We will document the modules requirement (via terraform-docs and any additional things we need to impart).

One of the things is where the module is expected to be used to access multiple accounts. We are writing the module for newer versions of Terraform, so the
```hcl
terraform {
  required_version = ">= 1.3.0"
  required_providers {
    aws = {
      source                = "hashicorp/aws"
      version               = "~> 5"
      configuration_aliases = [aws.current_account, aws.management_account]
    }
  }
}
```
is valid, and requires 2 providers to be supplied to allow the module to function as intended.

What that gives us though, is an error in tflint when run via pre-commit :
```
Terraform validate.......................................................Failed
- hook id: terraform_validate
- exit code: 1

Command 'terraform init' successfully done: .
Validation failed: .
╷
│ Error: Provider configuration not present
│ 
│ To work with aws_route53_vpc_association_authorization.this its original
│ provider configuration at
│ provider["registry.terraform.io/hashicorp/aws"].current_account is
│ required, but it has been removed. This occurs when a provider
│ configuration is removed while objects created by that provider still exist
│ in the state. Re-add the provider configuration to destroy
│ aws_route53_vpc_association_authorization.this, after which you can remove
│ the provider configuration again.
╵
╷
│ Error: Provider configuration not present
│ 
│ To work with aws_route53_zone_association.this its original provider
│ configuration at
│ provider["registry.terraform.io/hashicorp/aws"].management_account is
│ required, but it has been removed. This occurs when a provider
│ configuration is removed while objects created by that provider still exist
│ in the state. Re-add the provider configuration to destroy
│ aws_route53_zone_association.this, after which you can remove the provider
│ configuration again.
╵
```

What is the usual approach for using tflint on modules ... not full projects ... just the repo for the module that will be released with a semantic version tag.

All the other pre-commit hooks we run are fine. We can easily exclude the tflint check from the module repos (remove `      - id: terraform_tflint` from the `  - repo: https://github.com/antonbabenko/pre-commit-terraform` block).

Just don't want to do that and end up in a situation where there is a better solution.

Also, for the repos that do NOT have multiple providers, tflint runs successfully.

**Comment by rquadling** [comment]
Oh .. beep beep beep!!!

Ignore this. I'll leave the question up as an example of EXACTLY what RTFS means!

Sorry!!!

**Reply by 65156** [reply]
Would have been more helpful to post the answer for all the noobs visiting with the same issue...

**Reply by rquadling** [reply]
The very first 2 words of the output says it all.

In a way, a noob wouldn't have fallen foul as they WOULD RTFS as they are noobs!

It was a case of someone (me) thinking I knew it all and blamed the wrong tool. It's STILL me!

Sorry!

**Comment by bendrucker** [comment, marked as answer]
The issue was coming from `terraform validate` and not TFLint. The issue is https://github.com/hashicorp/terraform/issues/28490, and on GitHub Actions you can use https://github.com/bendrucker/terraform-configuration-aliases-action to automate the workaround.
